R/layer_population_scaling.R

```diff
@@ -134,24 +135,13 @@ slather.layer_population_scaling <-
     )
     rlang::check_dots_empty()
 
-    if (is.null(object$by)) {
-      object$by <- intersect(
-        kill_time_value(epi_keys(components$predictions)),
-        colnames(dplyr::select(object$df, !object$df_pop_col))
-      )
-    }
-    try_join <- try(
-      dplyr::left_join(components$predictions, object$df,
-        by = object$by
-      ),
-      silent = TRUE
+    object$by <- object$by %||% intersect(
+      kill_time_value(key_colnames(components$predictions)),
+      colnames(select(object$df, !object$df_pop_col))
     )
-    if (any(grepl("Join columns must be present in data", unlist(try_join)))) {
-      cli_stop(c(
-        "columns in `by` selectors of `layer_population_scaling` ",
-        "must be present in data and match"
-      ))
-    }
+    joinby <- list(x = names(object$by) %||% object$by, y = object$by)
+    hardhat::validate_column_names(components$predictions, joinby$x)
+    hardhat::validate_column_names(object$df, joinby$y)
```

Comment:
praise: thanks for handling this! this looks much better.

comment: not being familiar with the data objects here, i skimmed the unit tests for this layer and they look like they should catch any errors present, so i think this should be good to go.